### PR TITLE
Prepare to add k8s-server to Spring Initializr

### LINKED ISSUE #125: Change k8s-server POM to inherit from Spring Boot parent
As a developer, I'd like to change the k8s-server pom.xml to inherit the foundation bits from Spring Boot parent itself. This allows us to capture project metadata, so that we can publish the final artifacts to maven central.

### LINKED ISSUE #135: Move server configs to starter project
Any config files should move from the `spring-cloud-dataflow-server-kubernetes` to the `spring-cloud-starter-dataflow-server-kubernetes` subproject

### LINKED ISSUE #136: Add a BOM for kubernetes server

### PR DESCRIPTION
**Pull request by trisberg**
- Change k8s-server POM to inherit from Spring Boot parent

- Add a BOM for kubernetes server

- Move server configs to starter project

Part of #126

Resolves #135
Resolves #136
Resolves #125

### CODE REVIEW

File: spring-cloud-starter-dataflow-server-kubernetes/pom.xml [hunk 1]
```diff
@@ -21,5 +21,51 @@
 			<groupId>org.springframework.cloud</groupId>
 			<artifactId>spring-cloud-dataflow-server-core</artifactId>
 		</dependency>
+		<dependency>
```
Comment (ericbottard) [inline spring-cloud-starter-dataflow-server-kubernetes/pom.xml:24]: Shouldn't the versions setting for these deps be moved from the top-level pom to the BOM?

Comment (ericbottard) [reply spring-cloud-starter-dataflow-server-kubernetes/pom.xml:24]: Also, I've tested this with [this initializr change](https://github.com/ericbottard/initializr/commit/947d7b32be073b6791f936e2a53dc72875a2a213) (for which I'll be happy to issue a PR once RC1 is released if you want) and the generated project emits a wrong banner.
Need to add a `description` to this POM (as this is where the token replacement happens now):

```
  ____                              ____ _                __
 / ___| _ __  _ __(_)_ __   __ _   / ___| | ___  _   _  __| |
 \___ \| '_ \| '__| | '_ \ / _` | | |   | |/ _ \| | | |/ _` |
  ___) | |_) | |  | | | | | (_| | | |___| | (_) | |_| | (_| |
 |____/| .__/|_|  |_|_| |_|\__, |  \____|_|\___/ \__,_|\__,_|
  ____ |_|    _          __|___/                 __________
 |  _ \  __ _| |_ __ _  |  ___| | _____      __  \ \ \ \ \ \
 | | | |/ _` | __/ _` | | |_  | |/ _ \ \ /\ / /   \ \ \ \ \ \
 | |_| | (_| | || (_| | |  _| | | (_) \ V  V /    / / / / / /
 |____/ \__,_|\__\__,_| |_|   |_|\___/ \_/\_/    /_/_/_/_/_/

 Spring Cloud parent pom, managing plugins and dependencies for Spring Cloud projects 1.1.0.BUILD-SNAPSHOT

```

(I can add this on merge)

File: spring-cloud-dataflow-server-kubernetes-dependencies/pom.xml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	<artifactId>spring-cloud-dataflow-server-kubernetes-dependencies</artifactId>
+	<version>1.1.0.BUILD-SNAPSHOT</version>
+	<packaging>pom</packaging>
+	<name>spring-cloud-dataflow-server-kubernetes-dependencies</name>
+	<description>Spring Cloud Data Flow Server for Kunernetes Dependencies BOM designed to support consumption of Spring Cloud Data Flow Server for Kubernetes from the Spring Initializr.</description>
+	<dependencyManagement>
+		<dependencies>
+			<dependency>
+				<groupId>org.springframework.cloud</groupId>
+				<artifactId>spring-cloud-dataflow-server-kubernetes</artifactId>
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-kubernetes-dependencies/pom.xml]: I don't understand this.
Once a project is generated with initializr, it does **not** depend on `spring-cloud-dataflow-server-kubernetes` (as a matter of fact, a generated project is a sibling/competitor to `spring-cloud-dataflow-server-kubernetes`).
Why is this here?

Comment (ericbottard) [reply spring-cloud-dataflow-server-kubernetes-dependencies/pom.xml]: And, assuming my above comment is relevant, there would be only one depMgmt left in this bom. Does that make sense?

Comment (ericbottard) [reply spring-cloud-dataflow-server-kubernetes-dependencies/pom.xml]: ^^ the only point I see to having a single-dependency bom, is to have the ability to add [additional-boms](https://github.com/ericbottard/initializr/blob/AddDataFlowK8S/initializr-service/src/main/resources/application.yml#L51) to initializr.

But this seems more like a fundamental limitation of initializr (a checkbox there needs to map to only one dep) than a feature here.

File: spring-cloud-dataflow-server-kubernetes-dependencies/pom.xml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+				<version>1.1.0.BUILD-SNAPSHOT</version>
+			</dependency>
+			<dependency>
+				<groupId>org.springframework.cloud</groupId>
+				<artifactId>spring-cloud-starter-dataflow-server-kubernetes</artifactId>
+				<version>1.1.0.BUILD-SNAPSHOT</version>
+			</dependency>
+		</dependencies>
+	</dependencyManagement>
+	<distributionManagement>
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-kubernetes-dependencies/pom.xml:24]: I this actually useful/required? (Looks legit, but how is this used?)

File: spring-cloud-dataflow-server-kubernetes/pom.xml [hunk 4]
```diff
@@ -1,66 +1,107 @@
...
+		<dependencies>
+			<dependency>
+				<groupId>org.springframework.cloud</groupId>
+				<artifactId>spring-cloud-dataflow-server-kubernetes-dependencies</artifactId>
+				<version>1.1.0.BUILD-SNAPSHOT</version>
+				<type>pom</type>
+				<scope>import</scope>
+			</dependency>
+			<dependency>
+				<groupId>org.springframework.cloud</groupId>
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-kubernetes/pom.xml:84]: Why are we depending on this/these (spring-cloud) boms? Why aren't we depending on the spring-cloud-dataflow bom?

File: spring-cloud-dataflow-server-kubernetes/pom.xml [hunk 5]
```diff
@@ -1,66 +1,107 @@
...
+		<java.version>1.8</java.version>
+		<spring-cloud-task.version>1.1.0.RC1</spring-cloud-task.version>
+		<spring-cloud.version>Camden.SR2</spring-cloud.version>
 	</properties>
+
 	<dependencies>
 		<dependency>
 			<groupId>org.springframework.cloud</groupId>
 			<artifactId>spring-cloud-starter-dataflow-server-kubernetes</artifactId>
 			<version>1.1.0.BUILD-SNAPSHOT</version>
```
Comment (ericbottard) [inline spring-cloud-dataflow-server-kubernetes/pom.xml]: The version comes from the bom now

File: spring-cloud-starter-dataflow-server-kubernetes/pom.xml [hunk 6]
```diff
@@ -21,5 +21,51 @@
...
+			<artifactId>mariadb-java-client</artifactId>
+			<scope>runtime</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.postgresql</groupId>
+			<artifactId>postgresql</artifactId>
+			<scope>runtime</scope>
+		</dependency>
+		<!-- Fabric8 spring-cloud-kubernetes dependencies -->
+		<dependency>
```
Comment (ericbottard) [inline spring-cloud-starter-dataflow-server-kubernetes/pom.xml:35]: Actually, shouldn't the following 3 deps be in the deployer itself alone? Or are you suffering from the same problem as the CF-server wrt reactor (I doubt it)

File: spring-cloud-starter-dataflow-server-kubernetes/pom.xml [hunk 7]
```diff
@@ -21,5 +21,51 @@
...
+			</resource>
+		</resources>
+		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-resources-plugin</artifactId>
+			</plugin>
+			<plugin>
+				<groupId>pl.project13.maven</groupId>
+				<artifactId>git-commit-id-plugin</artifactId>
```
Comment (ericbottard) [inline spring-cloud-starter-dataflow-server-kubernetes/pom.xml]: This is not actually used, is it?

File: spring-cloud-dataflow-server-kubernetes-dependencies/pom.xml (hunk 2 above)
Comment (trisberg) [reply spring-cloud-dataflow-server-kubernetes-dependencies/pom.xml]: I was following the local server where the `spring-cloud-dataflow-server-local` is listed in the BOM. If it shouldn't be there and we can get by without a BOM, then that would be fine.

File: spring-cloud-dataflow-server-kubernetes-dependencies/pom.xml (hunk 3 above)
Comment (trisberg) [reply spring-cloud-dataflow-server-kubernetes-dependencies/pom.xml:24]: Don't know, copied from the local server

File: spring-cloud-dataflow-server-kubernetes/pom.xml (hunk 4 above)
Comment (trisberg) [reply spring-cloud-dataflow-server-kubernetes/pom.xml:84]: Was simply trying to mimic what initializr would generate - if we use different BOMs there then this can change

File: spring-cloud-starter-dataflow-server-kubernetes/pom.xml (hunk 6 above)
Comment (trisberg) [reply spring-cloud-starter-dataflow-server-kubernetes/pom.xml:35]: These are used by the server, not by the apps/deployer

File: spring-cloud-starter-dataflow-server-kubernetes/pom.xml (hunk 7 above)
Comment (trisberg) [reply spring-cloud-starter-dataflow-server-kubernetes/pom.xml]: Again, something that was used in the local starter, so it made it in here

File: spring-cloud-dataflow-server-kubernetes-dependencies/pom.xml (hunk 2 above)
Comment (ericbottard) [reply spring-cloud-dataflow-server-kubernetes-dependencies/pom.xml]: Well, my first comment definitely stands IMO: the server should NOT be there.

The question about the need for a BOM at all as a consequence is something else, maybe we should ask the boot guys

File: spring-cloud-dataflow-server-kubernetes/pom.xml (hunk 4 above)
Comment (ericbottard) [reply spring-cloud-dataflow-server-kubernetes/pom.xml:84]: I see. But initializr does that for local because of [this](https://github.com/ericbottard/initializr/blob/AddDataFlowK8S/initializr-service/src/main/resources/application.yml#L45). Seems more sensible to then depend on some "core" SCDF bom

File: spring-cloud-starter-dataflow-server-kubernetes/pom.xml (hunk 7 above)
Comment (ericbottard) [reply spring-cloud-starter-dataflow-server-kubernetes/pom.xml]: I think it should be removed, as I don't see it applied anywhere (no placeholder being used)

File: spring-cloud-starter-dataflow-server-kubernetes/pom.xml (hunk 6 above)
Comment (ericbottard) [reply spring-cloud-starter-dataflow-server-kubernetes/pom.xml:35]: Ah, correct.
As a sidenote, shouldn't they be `scope=runtime`?

File: spring-cloud-dataflow-server-kubernetes/pom.xml (hunk 5 above)
Comment (trisberg) [reply spring-cloud-dataflow-server-kubernetes/pom.xml]: Right, will remove

File: spring-cloud-starter-dataflow-server-kubernetes/pom.xml (hunk 1 above)
Comment (trisberg) [reply spring-cloud-starter-dataflow-server-kubernetes/pom.xml:24]: Even if we add the description, it looks like Boot grabs the first banner it finds, which seems to be the Data Flow core one - maybe we should move that one out of core and into the local starter

Comment (ericbottard) [reply spring-cloud-starter-dataflow-server-kubernetes/pom.xml:24]: yep, dataflow core should not have a banner IMO

Comment (ericbottard) [reply spring-cloud-starter-dataflow-server-kubernetes/pom.xml:24]: Also, I think it would be more correct to use `project.name` rather than `project.description` (and have `project.name` not be the lower case thing that `artifactId` already is

Comment (ericbottard) [reply spring-cloud-starter-dataflow-server-kubernetes/pom.xml:24]: OR, even crazier idea, just hardcode the damn thing

File: spring-cloud-starter-dataflow-server-kubernetes/pom.xml [hunk 8]
```diff
@@ -21,5 +21,51 @@
...
+					<include>META-INF/spring.factories</include>
+					<include>dataflow-server.yml</include>
+					<include>banner.txt</include>
+				</includes>
+			</resource>
+		</resources>
+		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-resources-plugin</artifactId>
```
Comment (ericbottard) [inline spring-cloud-starter-dataflow-server-kubernetes/pom.xml:65]: This plugin is part of the standard mvn config, AFAIK

File: spring-cloud-starter-dataflow-server-kubernetes/pom.xml (hunk 1 above)
Comment (trisberg) [reply spring-cloud-starter-dataflow-server-kubernetes/pom.xml:24]: Or do this - https://github.com/spring-cloud/spring-cloud-dataflow/issues/1008

That way we don't have to have a banner in every server impl

Comment (ericbottard) [reply spring-cloud-starter-dataflow-server-kubernetes/pom.xml:24]: That's another option, but again, which MANIFEST.MF will it read (there are jars in the jar!)

**Comment by ericbottard** [comment]
Squashed and merged as ba9cef3
